### PR TITLE
Deployment and code review of Stock Data App 1.0.0

### PR DESCRIPTION
**Pull request by eduardosancho**
# What was done in this PR: by @eduardosancho 🎃 

Thanks for taking the time to review the project. 
There are some __important things__ to note about it:

* It is based on the [Financial Modelling Prep API](https://site.financialmodelingprep.com/developer/docs), whose endpoints only provide information about a single company at a time, collective endpoints are a premium feature, thats why only 6 companies are displayed at the homepage. 💀 
* The chosen endpoints were the [Historical Stock Data](https://site.financialmodelingprep.com/developer/docs/historical-stock-data-free-api) which has the ability to be `filtered by date` and update the displayed content. 🔥 
* And the other endpoint is [Company Rating](https://site.financialmodelingprep.com/developer/docs#Company-Rating) which provides a summarized recommendation and rating about buying a particular stock. 🤑 

<hr />

### [Live Demo](https://sad-edison-40e613.netlify.app/)
image

<hr />

## Base Objectives:

- [x] Use `Function-Based React Components`. 🔧 

- [x] Use `React Props`. 📫 

- [x] Use `React Router`. 🪐 

- [x] Connect React and Redux 🌎 

- [x] Handle events using `reducers` and `dispatch`. ✋ 

- [x] Apply React best practices and language style guides in code. 📏 

- [x] Use correct Gitflow. 💧 

- [x] Follow best practices for HTML, and JavaScript. 🧑‍🎓 

- [x] Implement unit tests for all the reducers. ✔️ 

## Extra Objectives: 

- [x] Style the app using `React-BootStrap`. 💄 

- [x] Use `@fortawesome/react-fontawesome` for icons in the App. 🌞 

- [x] Use `Redux Toolkit` to handle events in the App. 🧰 

- [x] Use MSW to test the rendering of components by mocking a fetch to the API. ✅ 

<hr />

homer

Side note: This project was particularly challenging for me, and I'm proud of how far I got 😄 
``Again sincere thanks for the review!``

### CODE REVIEW

File: src/app/store.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { configureStore } from '@reduxjs/toolkit';
```
Comment (moise10r) [inline src/app/store.js]: - [x] Great job so far :+1:  However The structure of the files should follow the `ducks pattern`

eg. Has basic project structure

```
app-root-directory
  |- package.json
  |- README.md
  |- /public
    |- index.html
  |- /src
    |- app.css
    |- index.js
    |- /components
    |- /container
    |- /pages
    |- /redux
  |- /node_modules
```

File: src/index.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    'Ubuntu',
+    'Cantarell',
+    'Fira Sans',
+    'Droid Sans',
+    'Helvetica Neue',
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
```
Comment (moise10r) [inline src/index.css]: - [x] Kindly, you should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
- Typography: font face, size and weight.
- Layout: composition and space between elements. Kindly improve your design to match the original design and don't forget to add the missing `icon`.


||  Original [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))  ||
|-|-----------------------------------|-|
||go1||

||    `Home` page      ||      `Detail` page        ||
|-|----------------------|-| ----------------------|-|
||bad1 || bad2||

**Review by moise10r — CHANGES_REQUESTED** [review]
### Hi @eduardosancho,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
### To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Git flow is followed. :heavy_check_mark:
- Your README file looks awesome, the best one I've seen so far :+1: 
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Your app has been well deployed :+1: 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by eduardosancho** [comment]
Required changes have been executed on PR #6 
Mainly about the styling of the app and folder structure.

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @eduardosancho 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**